core/src/test/java/com/google/errorprone/refaster/UFreeIdentTest.java

```diff
@@ -41,10 +49,36 @@ public void inlinesExpression() {
 
   @Test
   public void binds() {
-    JCExpression expr = parseExpression("\"abcdefg\".charAt(x + 1)");
-    UFreeIdent ident = UFreeIdent.create("foo");
-    assertThat(ident.unify(expr, unifier)).isNotNull();
-    assertThat(unifier.getBindings()).containsExactly(new UFreeIdent.Key("foo"), expr);
+    CompilationTestHelper.newInstance(UnificationChecker.class, getClass())
+        .addSourceLines(
+            "A.java",
+            "class A {",
+            "  public void bar() {",
+            "    int x = 0;",
+            "    \"abcdefg\".charAt(x + 1);",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @BugPattern(
+      name = "UnificationChecker",
```

Comment:
After merging the PR, we can rebase it and also drop the `name` field here and below.